starter_code/src/movies.js

```diff
@@ -1,12 +1,37 @@
 /* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
-
+function turnHoursToMinutes(arr){ 
+  return arr.map(function(e,i,a){
+    if (e.duration.split(" ").length > 1) {
+      return {duration: +e.duration.split(" ")[0].slice(0,1)*60 + +e.duration.split(" ")[1].replace(/min/g,"")}
+    } 
+    else if (e.duration.split(" ").length == 1 && e.duration.includes("h")) {
+      return {duration: +e.duration.split(" ")[0].slice(0,1)*60}
+    }
+    else {
+      return {duration: +e.duration.replace(/min/g,"")}
+    }
+  });
+};
 
 // Get the average of all rates with 2 decimals 
-
+function ratesAverage(arr){ 
+  var total = arr.reduce(function(accumulator, current){
+    return accumulator + parseInt(current.rate) / arr.length;
+  },0);
+  return Math.round(total * 100)/100;  
+};
 
 // Get the average of Drama Movies
-
+function dramaMoviesRate(arr){
+  for(var i = 0; i < arr.length; i++){
+    arr.filter(function(e){
+      console.log(e.genre);
+      return(e.genre);
+    });
+  }
+ 
+}
 
 // Order by time duration, in growing order
 
```

Comment:
// Order by time duration, in growing order
function orderByDuration(arr) {
  arr.sort(function (a, b) {
    if (a.duration === b.duration) {
      if (a.title > b.title) {
        return 1;
      }
      if (a.title < b.title) {
        return -1;
      }
      return 0;
    }
    return a.duration - b.duration;
  });
  return arr;
}

// How many movies did STEVEN SPIELBERG
function howManyMovies(arr) {
  if (arr.length === 0) return;
  var drama = arr.filter(function (elem) {
    return elem.genre.includes('Drama');
  });
  var spielbergDramas = drama.filter(function (elem) {
    return elem.director.includes('Spielberg');
  });
  return 'Steven Spielberg directed ' + spielbergDramas.length + ' drama movies!';
}

// Order by title and print the first 20 titles
function orderAlphabetically(arr) {
  var result = [];
  var orderedArr = arr.sort(function (a, b) {
    if (a.title > b.title) {
      return 1;
    }
    if (a.title < b.title) {
      return -1;
    }
    return 0;
  });
  var limit = 20;
  if (orderedArr.length < 20) {
    limit = orderedArr.length;
  }
  for (var i = 0; i < limit; i++) {
    result.push(orderedArr[i].title);
  }
  return result;
}
// console.log(orderAlphabetically(movies));

// Best yearly rate average

function bestYearAvg(movies) {
  if (movies.length === 0) { return undefined; }

  var disctintYears = [];
  var aux = [];

  movies.forEach(function (e) {
    if (!aux.includes(e.year)) {
      aux.push(e.year);
      disctintYears.push({ year: e.year, rate: 0, films: 0 });
    }
  });

  disctintYears.forEach(function (y) {
    movies.forEach(function (e) {
      if (e.year === y.year) {
        if (!e.rate) {
          return;
        }
        y.rate += parseFloat(e.rate);
        y.films += 1;
      }
    });
    y.rate = parseFloat(y.rate / y.films);
  });

  disctintYears.sort(function (a, b) {
    if (a.rate === b.rate) {
      return a.year - b.year;
    }
    return b.rate - a.rate;
  });

  console.log(disctintYears);

  return 'The best year was ' + disctintYears[0].year + ' with an average rate of ' + disctintYears[0].rate;
}

Reply:
Muchas gracias Susana :-)